src/main/java/org/spongepowered/common/command/manager/SpongeCommandManager.java

```diff
@@ -493,49 +511,64 @@ private void prettyPrintThrowableError(final Throwable thr, final String command
+                        .warn("Command type '{}' has already been collected, skipping request from {}",
+                                handledType.toString(),
+                                type.getClass());
+            }
+        });
+        if (this.brigadierRegistrar == null) {
+            throw new IllegalStateException("Brigadier registrar was not detected");
+        }
+    }
+
+    private void registerInternalCommands(final CommandRegistrar<Parameterized> registrar) {
         try {
-            SpongeParameterizedCommandRegistrar.INSTANCE.register(
+            registrar.register(
                     Launch.getInstance().getCommonPlugin(),
                     this.spongeCommand.get().createSpongeCommand(),
                     "sponge"
-            );
+                                                                 );
         } catch (final CommandFailedRegistrationException ex) {
             throw new RuntimeException("Failed to create root Sponge command!", ex);
         }
         try {
             final PaginationService paginationService = Sponge.getServiceProvider().paginationService();
             if (paginationService instanceof SpongePaginationService) {
-                SpongeParameterizedCommandRegistrar.INSTANCE.register(
+                registrar.register(
                         Launch.getInstance().getCommonPlugin(),
                         ((SpongePaginationService) paginationService).createPaginationCommand(),
                         "pagination", "page"
-                );
+                                                                     );
             }
         } catch (final CommandFailedRegistrationException ex) {
             throw new RuntimeException("Failed to create pagination command!", ex);
         }
-        final Set<TypeToken<?>> usedTokens = new HashSet<>();
-        Sponge.getGame().registries().registry(RegistryTypes.COMMAND_REGISTRAR).streamEntries().forEach(entry -> {
-            final CommandRegistrar<?> registrar = entry.value();
-            // someone's gonna do it, let's not let them take us down.
-            final TypeToken<?> handledType = registrar.handledType();
-            if (handledType == null) {
-                SpongeCommon.getLogger().error("Registrar '{}' did not provide a handledType, skipping...", registrar.getClass());
-            } else if (usedTokens.add(handledType)) { // we haven't done it yet
-                this.game.getEventManager().post(this.createEvent(cause, this.game, registrar));
-            } else {
-                SpongeCommon.getLogger()
-                        .warn("Command type '{}' has already been collected, skipping request from {}",
-                                handledType.toString(),
-                                registrar.getClass());
-            }
-        });
-        SpongeParameterizedCommandRegistrar.INSTANCE.register(
+
+        // TODO: this used to be done after the event, do we care?
```

Comment:
It doesn't matter. Callbacks in Sponge should be calling `sponge:callback` anyway, so as long as that registers, it should be fine.